.github/workflows/changelog-check.yml

```diff
@@ -0,0 +1,39 @@
+# The outline of this workflow is something that the GitHub Security team warns against
+# here: https://securitylab.github.com/research/github-actions-preventing-pwn-requests. But
+# due to this workflow only running a git diff check and not building or publishing anything,
+# there is no harm in checking out the PR HEAD code.
```

Comment:
I think the most likely risk here would be someone moving this logic into a script, and calling the script from this workflow.

I think it might help to clarify this point with something like this:


> All the code checked out in this workflow should be considered untrusted. This workflow must never call any makefiles or scripts. It must never be changed to run any code from the checkout.

Reply:
That's true. The only thing that gets 'written' back is a PR comment but the copied script on a PR wouldn't have access to the github token to do that.

That blurb is nice to have though, I'll add it!